platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/DefaultIvyArtifactRepository.java

```diff
@@ -248,17 +255,25 @@ private MetaDataParser<MutableIvyModuleResolveMetadata> createIvyDescriptorParse
     }
 
     @Override
-    public URI getUrl() {
+    @ReplacesEagerProperty(adapter = DefaultIvyArtifactRepositoryAdapter.class)
+    public Property<URI> getUrl() {
         return urlArtifactRepository.getUrl();
     }
 
+    @Override
+    @Deprecated
+    public void setUrl(Object url) {
```

Comment:
❓ Why we kept `setUrl(Object url)` here, why we don't handle translation in the `Property`?

Reply:
Mostly because it's easier to keep it until RepositoryHandlerExtensions.kt are deprecated. They don't have access to fileResolver.